pkg/controller/nodeipam/ipam/multi_cidr_priority_queue.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ipam
```

Comment:
It probably makes sense in the future to put independent utilities like this in a PR by itself, but reference the overall API change PR so it can be more easily reviewed and merged.

Reply:
For now is it ok if I move the utilities to separate commits in this PR, as it would preserve the review comments